### ISSUE TITLE
Remove bias from Conv op schema?

### ISSUE THREAD

**Issue by rryan** [onnx/onnx#299]
I noticed that the Conv op has an optional bias baked in. This strikes me as an unnecessary coupling of two related concepts. If a bias is included, why not also include an activation, and batch norm, and ?

**Comment by ebarsoum** [comment]
This fused OP similar to GEMM and others are for performance.

**Comment by rryan** [comment]
Maybe this is more of a philosophical point, but AIUI ONXX is meant to be a universal IR for neural networks?

I would expect fusing the conv and a bias following it (if present) is the job of the compiler backend.

**Comment by ezyang** [comment]
Right, so gemm and convolution are two cases where we've compromised on philosophical purity in order to pave the cowpaths for the most common implementations. Most backends are using CuDNN which do fuse, which means that if you didn't fuse here, you are basically mandating that every backend implement the same optimization pass, repeatedly, to recover information which you deleted at export time. This is not good design; don't make it hard to do the right thing.

**Comment by linkerzhang** [comment]
+1 on "even be implemented as optimization passes in ONNX that can be selectively invoked by individual backends", this may resolve the issue of mandating each backend to have same optimization implementation and keep the philosophical purity.
